src/pods/user-settings/user-settings.component.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import {
+  Typography,
+  Checkbox,
+  FormControlLabel,
+  Button,
+  TextField,
+  Container,
+} from '@material-ui/core';
+import { UserSettings } from './user-settings.vm';
+import produce from 'immer';
+
+interface Props {
+  onSave: (settings: UserSettings) => void;
+  onCancel: () => void;
+  userSettings: UserSettings;
+}
+
+export const UserSettingsComponent: React.FC<Props> = props => {
+  const { onSave, onCancel, userSettings } = props;
+  const [temporalSettings, setTemporalSettings] = React.useState(userSettings);
+
+  React.useEffect(() => {
+    setTemporalSettings(userSettings);
+  }, [userSettings]);
+
+  const handleSecondChanceChange = () => (
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setTemporalSettings(
+      produce(temporalSettings, draft => {
+        draft.secondChance = !draft.secondChance;
+      })
+    );
+  };
+
+  const handleNumberQuestionsChange = () => (
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setTemporalSettings(
+      produce(temporalSettings, draft => {
+        draft.numberQuestions = e.target.valueAsNumber;
+      })
+    );
+  };
+
+  return (
+    <>
+      <Container>
+        {temporalSettings ? (
```

Comment:
I think we can remove temporalSettings and ensure it will always come informed (we can even setup a defualt property value in case is not informed)